### PR TITLE
Implement OwnEndpoint such that query returns own IP

### PR DESCRIPTION
**Pull request by onkarbanerjee**

### CODE REVIEW

File: contrib/coredns/policy/policy.go [hunk 1]
```diff
@@ -92,6 +93,45 @@ func (p *policyPlugin) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dn
 		p.attrPool.Put(attrBuffer)
 	}()
 
+	if p.conf.ownIPEndpoint != "" {
+		var buff bytes.Buffer
+		buff.Write([]byte(p.conf.ownIPEndpoint))
+		buff.Write([]byte("."))
+		buff.Write([]byte(p.conf.debugSuffix))
```
Comment (onkarbanerjee) [inline contrib/coredns/policy/policy.go]: Any better way to do this, maybe?

Comment (rdrozhdzh) [reply contrib/coredns/policy/policy.go]: concatenate it once (in parsing code) then store in config

File: contrib/coredns/policy/policy.go [hunk 2]
```diff
@@ -92,6 +93,45 @@ func (p *policyPlugin) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dn
...
+
+		if r != nil && len(r.Question) > 0 && r.Question[0].Name == buff.String() {
+			var rr dns.RR
+			srcIP := getRemoteIP(w)
```
Comment (rdrozhdzh) [inline contrib/coredns/policy/policy.go]: create `func ip2rr(ip net.IP) dns.RR` which will create A or AAAA RR from IP address. This func should be reused in `setRedirectQueryAnswer()`

File: contrib/coredns/policy/policy.go [hunk 3]
```diff
@@ -92,6 +93,45 @@ func (p *policyPlugin) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dn
 		p.attrPool.Put(attrBuffer)
 	}()
 
+	if p.conf.ownIPEndpoint != "" {
+		var buff bytes.Buffer
+		buff.Write([]byte(p.conf.ownIPEndpoint))
+		buff.Write([]byte("."))
+		buff.Write([]byte(p.conf.debugSuffix))
+
+		if r != nil && len(r.Question) > 0 && r.Question[0].Name == buff.String() {
```
Comment (rdrozhdzh) [inline contrib/coredns/policy/policy.go]: move this code to a separate func, keep ServeDNS as short as possible

Comment (rdrozhdzh) [inline contrib/coredns/policy/policy.go]: do you care about query type and query class?

**Pull request closed without merging**